dds/src/Publisher.cpp

```diff
@@ -32,17 +32,22 @@ namespace dds {
 Publisher::Publisher(
         Participant* participant,
         const std::string& topic_name,
-        const std::string& message_type)
+        const ::xtypes::DynamicType& message_type)
 
     : topic_name_{topic_name}
-    , message_type_{message_type}
 {
-    dynamic_data_ = participant->create_dynamic_data(message_type);
+    DynamicTypeBuilder* builder = Conversion::create_builder(message_type);
+    if (builder != nullptr)
+    {
+        participant->register_dynamic_type(topic_name, builder);
+    }
+
+    dynamic_data_ = participant->create_dynamic_data(topic_name);

```

Comment:
Why use the `topic_name` instead of the `message_type.name()`?

Reply:
Nice catch, it's a potential bug!

Reply:
What happen if the user has two topic with the same type?

Reply:
Everything will work as expected. The map will contain two entries, both pointing to the same topic. 
My previous comment was confusing, the relationship is N - 1. A topic uses only one type, but a type could be used by any number of topics.

Reply:
Ok!